builder/oracle/bmcs/config.go

```diff
@@ -0,0 +1,216 @@
+		accessCfgFile = c.AccessCfgFile
+	} else {
+		accessCfgFile, err = getDefaultBMCSSettingsPath()
+		if err != nil {
+			accessCfgFile = "" // Access cfg might be in template
+		}
+	}
+
+	accessCfg := &client.Config{}
+
+	if accessCfgFile != "" {
+		loadedAccessCfgs, err := client.LoadConfigsFromFile(accessCfgFile)
+		if err != nil {
+			return nil, fmt.Errorf("Invalid config file %s: %s", accessCfgFile, err)
+		}
+		cfgAccount := "DEFAULT"
+		if c.AccessCfgFileAccount != "" {
+			cfgAccount = c.AccessCfgFileAccount
+		}
+
+		var ok bool
+		accessCfg, ok = loadedAccessCfgs[cfgAccount]
+		if !ok {
+			return nil, fmt.Errorf("No account section '%s' found in config file %s", cfgAccount, accessCfgFile)
+		}
+	}
+
+	// Override SDK client config with any non-empty template properties
+
+	if c.UserID != "" {
+		accessCfg.User = c.UserID
+	}
+
+	if c.TenancyID != "" {
+		accessCfg.Tenancy = c.TenancyID
+	}
+
+	if c.Region != "" {
+		accessCfg.Region = c.Region
+	} else {
+		accessCfg.Region = "us-phoenix-1"
+	}
+
+	if c.Fingerprint != "" {
+		accessCfg.Fingerprint = c.Fingerprint
+	}
+
+	if c.PassPhrase != "" {
+		accessCfg.PassPhrase = c.PassPhrase
+	}
+
+	if c.KeyFile != "" {
+		accessCfg.KeyFile = c.KeyFile
+		accessCfg.Key, err = client.LoadPrivateKey(accessCfg)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to load private key %s : %s", accessCfg.KeyFile, err)
+		}
+	}
+
+	var errs *packer.MultiError
+	if es := c.Comm.Prepare(&c.ctx); len(es) > 0 {
+		errs = packer.MultiErrorAppend(errs, es...)
+	}
+
+	// Required AccessCfg configuration options
+
+	if accessCfg.User == "" {
+		errs = packer.MultiErrorAppend(
+			errs, errors.New("'user_ocid' must be specified"))
+	}
+
+	if accessCfg.Tenancy == "" {
+		errs = packer.MultiErrorAppend(
+			errs, errors.New("'tenancy_ocid' must be specified"))
+	}
+
+	if accessCfg.Region == "" {
+		errs = packer.MultiErrorAppend(
+			errs, errors.New("'region' must be specified"))
+	}
+
+	if accessCfg.Fingerprint == "" {
+		errs = packer.MultiErrorAppend(
+			errs, errors.New("'fingerprint' must be specified"))
+	}
+
+	if accessCfg.Key == nil {
+		errs = packer.MultiErrorAppend(
+			errs, errors.New("'key_file' must be specified"))
+	}
+
+	c.AccessCfg = accessCfg
+
+	// Required non AccessCfg configuration options
+
+	if c.AvailabilityDomain == "" {
+		errs = packer.MultiErrorAppend(
+			errs, errors.New("'availability_domain' must be specified"))
+	}
+
+	if c.CompartmentID == "" {
+		c.CompartmentID = accessCfg.Tenancy
+	}
+
+	if c.Shape == "" {
+		errs = packer.MultiErrorAppend(
+			errs, errors.New("'shape' must be specified"))
+	}
+
+	if c.SubnetID == "" {
+		errs = packer.MultiErrorAppend(
+			errs, errors.New("'subnet_ocid' must be specified"))
+	}
+
+	if c.BaseImageID == "" {
+		errs = packer.MultiErrorAppend(
+			errs, errors.New("'base_image_ocid' must be specified"))
+	}
+
+	if c.ImageName == "" {
+		errs = packer.MultiErrorAppend(
+			errs, errors.New("'image_name' must be specified"))
+	}
```

Comment:
This should have a default value instead of being required, Defaults to `packer-{{timestamp}}`.